### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/opengeode/commit/96af1593e5aee09f4fb3a19a9bc147d9c56d0c76#commitcomment-77030445

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/opengeode/commit/0250e4488c2ce45ed4a2ed9385b5b2dc2763c6a0

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

I will be annotating this PR, and I fully expect to be asked to drop certain files.

### CODE REVIEW

File: opengeode/AdaGenerator.py [hunk 1]
```diff
@@ -528,7 +528,7 @@ def _process(process, simu=False, instance=False, taste=False, **kwargs) -> str:
         ads_template.append(f'procedure Startup;')
 
 
-    # Generate the the code of the procedures
+    # Generate the code of the procedures
```
Comment (jsoref) [inline opengeode/AdaGenerator.py:531]: I'm still playing w/ duplicate word detection. I think this class of changes is correct.

File: opengeode/ogParser.py [hunk 2]
```diff
@@ -921,7 +921,7 @@ def check_one_param(p, name):
         # When we get there, the parameter has been checked already and we
         # know it is a CHOICE type.
         # However it may be a subtype, and in the AST the choices are defined
-        # in the supertype, so we must find it and and the -selection suffix
+        # in the supertype, so we must find it and add the -selection suffix
```
Comment (jsoref) [inline opengeode/ogParser.py:924]: This is an atypical case of duplicate word detection...

File: opengeode/Helper.py [hunk 3]
```diff
@@ -608,7 +608,7 @@ def _rename_primcall(ast, from_name, to_name):
 
 
 @rename_everything.register(ogAST.PrimConditional)
-def _rename_ifhthenelse(ast, from_name, to_name):
+def _rename_ifthenelse(ast, from_name, to_name):
```
Comment (jsoref) [inline opengeode/Helper.py:611]: ??

File: opengeode/opengeode.py [hunk 4]
```diff
@@ -614,10 +614,10 @@ def recursive_render(content, dest_scene):
                     try:
                         if s.ast.errors or s.ast.warnings:
                             # Keep track of all symbols containing errors,
-                            # they are processesd in "find_symbols_and_update_errors"
+                            # they are processed in "find_symbols_and_update_errors"
                             G_ERRORS.append(s)
                     except Exception as e:
-                        print("Ooops", str(e))
+                        print("Oops", str(e))
```
Comment (jsoref) [inline opengeode/opengeode.py:620]: This repository doesn't appear to be very "cute", so there didn't seem to be a big reason not to make this change.

File: opengeode/opengeode.py [hunk 5]
```diff
@@ -1069,7 +1069,7 @@ def paste_symbols(self):
 
     def sdl_to_statechart(self, basic=True, view=None, ast=None):
         ''' Create a graphviz representation of the SDL model 
-            Optionally take a QGraphicsView to use as parent for modals '''
+            Optionally take a QGraphicsView to use as parent for models '''
```
Comment (jsoref) [inline opengeode/opengeode.py:1072]: I'm not terribly confident about this. `modal` in relation to Qt could refer to a modal dialog...

File: opengeode/sdlSymbols.py [hunk 6]
```diff
@@ -1132,7 +1132,7 @@ def conn_end_zones(self):
         yield QRect(15, 5, rect.width() - 30, 10)
         yield QRect(5, 5, 10, rect.height() - 10)
         yield QRect(rect.width() - 15, 5, 10, rect.height() - 10)
-        yield QRect(15, rect.heigth() - 15, rect.width() - 30, 10)
+        yield QRect(15, rect.height() - 15, rect.width() - 30, 10)
```
Comment (jsoref) [inline opengeode/sdlSymbols.py:1135]: ???

File: tests/testsuite/TrafficLight_Basic/Makefile [hunk 7]
```diff
@@ -44,7 +44,7 @@ test-c:
 
 
 test-llvm:
-	$(OPENGEODE) controller.pr system_strucure.pr --llvm -O$(O)
+	$(OPENGEODE) controller.pr system_structure.pr --llvm -O$(O)
```
Comment (jsoref) [inline tests/testsuite/TrafficLight_Basic/Makefile:47]: ??

File: tests/testsuite/test-eds1/dataview.asn [hunk 8]
```diff
@@ -111,7 +111,7 @@ MASInterfaceTypeClientType ::= SEQUENCE
 }
 
 
--- message types for DemoDeviceDSAPSubnetworkps
+-- message types for DemoDeviceDSAPSubnetworkPS
```
Comment (jsoref) [inline tests/testsuite/test-eds1/dataview.asn:114]: This matches some other similar use elsewhere...

File: tests/testsuite/test-equal/og.pr [hunk 9]
```diff
@@ -16,7 +16,7 @@ process og;
     dcl V6 Toto := { elem_1 'Hello', elem_2 {'Ola', 'TheWorld'} };
     
     dcl V7 AChoice := a: TRUE;
-    dcl V8 ACHoice := b: {'Hello', 'World!'};
+    dcl V8 AChoice := b: {'Hello', 'World!'};
```
Comment (jsoref) [inline tests/testsuite/test-equal/og.pr:19]: The other items preceding this reuse the left-hand-side of the `:=` operator..

File: tests/testsuite/test-expressions/expressions.pr [hunk 10]
```diff
@@ -113,7 +113,7 @@ process expressions;
             /* CIF task (401, 2300), (263, 35) */
             task varSeqOf1 := varSeqOf1 // varSeqOf2;
             /* CIF PROCEDURECALL (351, 2350), (363, 35) */
-            call assert(length(varSeqOf1) = 6, 'lenght(varSeqOf1) = 6');
+            call assert(length(varSeqOf1) = 6, 'length(varSeqOf1) = 6');
```
Comment (jsoref) [inline tests/testsuite/test-expressions/expressions.pr:116]: ?

File: tests/testsuite/test-ibk/dataview-uniq.asn [hunk 11]
```diff
@@ -166,7 +166,7 @@ MyL3DstAddr ::= My4BitArray
 -- L3 FRAME: Control - frame type
 MyL3Cntrl ::= My8BitArray -- Fixed value 0x03
 
--- L3 FRAME: Code - data frame identificator
+-- L3 FRAME: Code - data frame identifier
```
Comment (jsoref) [inline tests/testsuite/test-ibk/dataview-uniq.asn:169]: It's possible this is a term of art.

**Review by jsoref — COMMENTED** [review]
Most corrections were suggested by Google Sheets. All fault is mine.

**Review by maxime-esa — APPROVED** [review]
SDL is not case sensitive, but consistency fix appreciated

**Review by maxime-esa — CHANGES_REQUESTED** [review]
"actual", not "actually"

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — CHANGES_REQUESTED** [review]
This one is not right. "delem" is meant for "dash-element" as it results from elem.replace('_', '-')

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — APPROVED** [review]

**Review by maxime-esa — CHANGES_REQUESTED** [review]
This one is not right. The "Emmit" typo is from another tool (asn1scc). If we change the template headers here, that will break.

File: opengeode/AdaGenerator.py [hunk 12]
```diff
@@ -2526,9 +2526,9 @@ def _sequence(seq, **kwargs):
     absent_fields = []
     for elem, value in seq.value.items():
         # Set the type of the field - easy thanks to ASN.1 flattened AST
-        delem = elem.replace('_', '-')
```
Comment (maxime-esa) [inline opengeode/AdaGenerator.py:2529]: No, keep "delem" here (dash element)

File: opengeode/CGenerator.py [hunk 13]
```diff
@@ -1978,10 +1978,10 @@ def _sequence(seq):
     absent_fields = []
     for elem, value in seq.value.items():
         # Set the type of the field - easy thanks to ASN.1 flattened AST
-        delem = elem.replace('_', '-')
+        delim = elem.replace('_', '-')
```
Comment (maxime-esa) [inline opengeode/CGenerator.py:1981]: Keep delem here too

File: opengeode/Helper.py (hunk 3 above)
Comment (maxime-esa) [reply opengeode/Helper.py:611]: Name of the function does not matter here (singledispatch is used) but should indeed be fixed.

File: opengeode/StgBackend.py [hunk 14]
```diff
@@ -1296,9 +1296,9 @@ def _sequence(seq):
     absent_fields = []
     for elem, value in seq.value.items():
         # Set the type of the field - easy thanks to ASN.1 flattened AST
-        delem = elem.replace('_', '-')
+        delim = elem.replace('_', '-')
```
Comment (maxime-esa) [inline opengeode/StgBackend.py:1299]: Keep delem

File: opengeode/opengeode.py (hunk 5 above)
Comment (maxime-esa) [reply opengeode/opengeode.py:1072]: Indeed `modals` is what is meant here.

File: opengeode/sdlSymbols.py (hunk 6 above)
Comment (maxime-esa) [reply opengeode/sdlSymbols.py:1135]: Ouch, that is a clear bug!

File: tests/testsuite/test-expressions/expressions.pr (hunk 10 above)
Comment (maxime-esa) [reply tests/testsuite/test-expressions/expressions.pr:116]: It's in a comment, so harmless

File: tests/testsuite/TrafficLight_Basic/Makefile (hunk 7 above)
Comment (maxime-esa) [reply tests/testsuite/TrafficLight_Basic/Makefile:47]: This would indeed break the tests, but the LLVM backend is not used anymore.

**Review by maxime-esa — CHANGES_REQUESTED** [review]
Thanks a lot for this spell checking work!
There are only 3 cases which need to remain as is:
* the `Emmit` typo in the StringTemplate headers
* `modals`
*  `delem` (at various places)

**Review by maxime-esa — APPROVED** [review]
Look all good, thanks!

**Pull request merged**